src/Shared-Src/Native/loader.cpp

```diff
@@ -79,13 +73,26 @@ namespace shared {
     };
 
     void Loader::CreateNewSingeltonInstance(ICorProfilerInfo4* pCorProfilerInfo,
-                                            std::function<void(const std::string& str)> logDebugCallback,
-                                            std::function<void(const std::string& str)> logInfoCallback,
-                                            std::function<void(const std::string& str)> logWarnCallback,
-                                            const LoaderResourceMonikerIDs& resourceMonikerIDs,
-                                            WCHAR const * native_profiler_library_filename)
+                                            std::function<void(const std::string& str)> log_debug_callback,
+                                            std::function<void(const std::string& str)> log_info_callback,
+                                            std::function<void(const std::string& str)> log_warn_callback,
+                                            const LoaderResourceMonikerIDs& resource_moniker_ids,
+                                            WCHAR const * native_profiler_library_filename,

```

Comment:
Maybe in another review:
Wrap all of this in an object (struct/class)
Same for logging functions
What do you think?

Reply:
Good point. I prefer to look into it once we know how the logging stull will look like long term. ))